include/ncengine/ecs/Component.h

```diff
@@ -51,11 +52,17 @@ class FreeComponent : public ComponentBase
         #endif
 };
 
+/** @brief Type representation of a null component. */
+struct NullComponent : public ComponentBase
```

Comment:
Are we using this anywhere but the tests? What is the use case for this other than testing?

Reply:
Yes... looking... ok its just in tests here, the other uses haven't been checked in, but I'll need it at least for #438. Its intended for template contexts where you need to sub in some component type as a template argument, but don't know the concrete component type being processed, like with `AnyComponent`.

You can do stuff like this this:
```cpp
// Where we don't know the component type
// Get size/alignment info for an arbitrary `AnyImplConcrete<>` and declare storage
using Derived = AnyImplConcrete<NullComponent>;
alignas(alignof(Derived)) std::byte buffer[sizeof(Derived)];

// Where we do know the component type
// Construct real impl on the buffer (template substitution must not alter characteristics)
new (buffer) AnyImplConcrete<ActualType>(instance, handler);

// Back where we don't know the type
// Cast to base ptr and happily call any virtual funcs
reinterpret_cast<AnyImplBase*>(buffer)->DrawUI();
```

Reply:
Ok, I think this can actually be removed now that #442 is merged in. This was only necessary to satisfy the `std::derived_from<ComponentBase>` part of `PooledComponent` - any type can serve as a stand-in now.

Reply:
Ok, removed